ibis/pandas/execution/tests/test_localcontext.py

```diff
@@ -0,0 +1,119 @@
+import pandas as pd
+import pandas.util.testing as tm
+import pytest
+
+import ibis
+import ibis.expr.operations as ops
+from ibis.pandas.client import PandasClient, PandasTable
+from ibis.pandas.dispatch import (
+    compute_local_context,
+    is_computable_input,
+    pre_execute,
+)
+
+pytestmark = pytest.mark.pandas
+
+
+def test_execute_with_localcontext(t):
+    # define a pre_execute function that trim data accroding to localcontext
+    @pre_execute.register(PandasTable, PandasClient)
+    def execute_table_with_localcontext(
+        op, client, scope=None, localcontext=None, **kwargs
+    ):
+        # retreive df from dict
+        df = client.dictionary[op.name]
+        try:
+            begin, end = map(pd.to_datetime, localcontext)
+        except ValueError:
+            return {op: df}
+        # filter time col with time context
+        time_col = "plain_datetimes_naive"
+        return {op: df[df[time_col] >= begin][df[time_col] < end]}
+
+    expr = t
+    # define a time context for time-series data
+    context = ("20170101", "20170103")
+
+    # without time context, execute produces every row
+    df_all = expr.execute()
+    assert len(df_all["plain_datetimes_naive"]) == 3
+
+    # with context set, execute produces only rows within context
+    df_within_context = expr.execute(localcontext=context)
+    assert len(df_within_context["plain_datetimes_naive"]) == 1
+
+    del pre_execute.funcs[(PandasTable, PandasClient)]
+    pre_execute.reorder()
+    pre_execute._cache.clear()
+
+
+merge_asof_minversion = pytest.mark.skipif(
+    pd.__version__ < '0.19.2',
+    reason="at least pandas-0.19.2 required for merge_asof",
+)
+
+
+@merge_asof_minversion
+def test_context_adjustment(
```

Comment:
Can you add a rule/test for time window as well?

Reply:
Added in ``timecontext.py``